### DISCUSSION TITLE
[BP#34] Redimensionner les images en dehors du HTML

### DISCUSSION THREAD

**Discussion by Fred-artwai** [cnumr/best-practices#239] [Bonnes pratiques] [answered]
Je suis très dubitatif sur cette BP.
D'une part, elle fait fi des images responsives. Une image dans un site dispose de plusieurs formats, de plusieurs tailles envoyées au navigateurs selon son écran, et le support du format.
D'autre part, je n'ose imaginer tous les webmasters du monde faire "a la main" tout ces formats. De plus en achetant une license Photoshop... 
Enfin, il y a désormais pleins d'outils cotés serveur avec ou sans CDN pour s'occuper de l'optimisation des images pour n'envoyer que le format le plus adapté au device qui le demande.

J'opterais pour une formulation plus user-friendly : "Ne pas redimensionner les images cotés navigateur"
Description outrageusement piqué à Dareboost :
"Si vos images sont plus grandes que leur zone d'affichage, le navigateur téléchargera des données inutiles et procédera à des redimensionnements disgracieux."

Qu'en pensez-vous ?

**Comment by ACTLEM** [comment]
> J'opterais pour une formulation plus user-friendly : "Ne pas redimensionner les images cotés navigateur"

J'ai toujours compris cette BP comme cela. Donc +1 pour clarifier le titre de cette BP.
Effectivement il serait plus opportun de parler d'utilisation de DAM ou des outils pour mettre à disposition plusieurs formats d'une même image.

**Comment by LaFelixe** [comment]
Je suis d'accord que le passage sur la méthode Photoshop est un peu daté.
Mais je ne vois pas en quoi la BP fait fi des images responsives. Ces multiples dimensions d'image sont bien sur le serveur et chargées à la demande comme tu le décris. Elles ne sont pas redimensionnées dans le navigateur via le HTML ou le CSS. La méthode est donc bien conforme à la BP, il n'y a pas lieu d'en parler à mon avis.

**Reply by tbroyer** [reply]
> Mais je ne vois pas en quoi la BP fait fi des images responsives. Ces multiples dimensions d'image sont bien sur le serveur et chargées à la demande comme tu le décris. Elles ne sont pas redimensionnées dans le navigateur via le HTML ou le CSS. La méthode est donc bien conforme à la BP, il n'y a pas lieu d'en parler à mon avis.

Oui et non. Si je mets une image en width:100%, background-size:cover ou équivalent, même si on va choisir l'image à la dimension la plus proche, on va quand même la redimensionner dans le navigateur. Il faudrait probablement ajouter une note dans la BP.

**Comment by ACTLEM** [comment, marked as answer]
J'ai créé une issue pour clarifier le titre et ajouter une note sur le redimensionnement: https://github.com/cnumr/best-practices/issues/259

**Reply by Fred-artwai** [reply]
@ACTLEM j'ai un doute quand il y a une issue qui m'est affectée, je met où le texte qui m'incombe ? Je suis un peu perdu là ?

**Reply by ACTLEM** [reply]
Le mieux c'est de créer une branche (pas la main) et une Pull Request pour review. Dans l'issue il y a le nom du fichier de la bonne pratique à modifier ou à ajouter (ou à supprimer).
Si tu n'es pas à l'aise, tu peux mettre dans l'issue les changements effectifs que tu vois et quelqu'un d'autre créera la Pull Request pour toi.

**Reply by Fred-artwai** [reply]
Je vais faire la deuxième c'est plus simple pour moi. Merci.
